modules/drivers/druid/src/metabase/driver/druid/client.clj

```diff
@@ -107,23 +107,18 @@
   [canceled-chan details query]
   {:pre [(map? details) (map? query)]}
   (let [query-id  (get-in query [:context :queryId])
-        query-fut (future
-                    (try
-                      (do-query details query)
-                      (catch Throwable e
```

Comment:
Derefing a future below conveys the exceptions thrown inside the future, so we don't really need to catch and return exception here only to throw it again below.